docs/best_practices.md

```diff
@@ -0,0 +1,153 @@
+
+However, if you create a unique Label to identify those resources, you can use that when exporting so that you don't have to know the IDs of each individual resource you want to include, like this:
+
+```
+influx pkg export all --filter=labelName=your_label_name
+```
+
+## Telegraf Configurations
+
+A good Template needs to be more than just a Dashboard, it should also include ways to send data to a user's InfluxDB instance. This can be done in a number of ways, from making calls directly to the [InfluxDB API](https://v2.docs.influxdata.com/v2.0/write-data/#influxdb-api), using the [InfluxDB Client libraries](https://v2.docs.influxdata.com/v2.0/reference/api/client-libraries/), or by [using Telegraf](https://v2.docs.influxdata.com/v2.0/write-data/use-telegraf/) and it's many plugins that do the bulk of the work for you.
+
+Telegraf is a widely used data collection agent with a large community that has contributed plugins for gathering data from a variety of sources. Whatever data you're collecting, chances are there's a Telegraf plugin that can help you do it. As such we encourage Templates to include a Telegraf configuration for data collection. If you use another method of sending data to InfluxDB, please include that and instructions on using it in your Template's `README.md`.
+
+### Using Environment Variables
+
+In order to make your configuration usable by others, certain settings inside it will need to change. Fortunately Telegraf makes this easy by looking for variables in your configuration and replacing them with [environment variables](https://github.com/influxdata/telegraf/blob/master/docs/CONFIGURATION.md#environment-variables). Then users can run your Telegraf configuration unmodified by simply setting those values in their environment before running it.
+
+The main settings you'll want to replace with variables are in the `[[outputs.influxdb_v2]]` section of your configuration:
+
+```
+urls = ["$INFLUX_HOST"]
+
+token = "$INFLUX_TOKEN"
+
+organization = "$INFLUX_ORG"
+```
+
+If your template allows the user to choose their own `bucket` values, you should make this a variable in your configuration also:
+
+```
+bucket = "$INFLUX_BUCKET"
+```
+
+Depending on what input plugins you use, there will likely be others that you want to make variables, such as host names, databases, or authentication credentials.
+
+> **Note:** Be sure to document all of these variables in the `README.md` for your Template so that users will know what they need to set and what values they should use.
+
+### Telegraf Configuration Labels
+
+Sometimes, it's handy to group Telegraf connfigurations by specific plugins. For example, if I want to make a change to an input plugin, being able to quickly find all the Telegraf configurations using that plugin is helpful. To this end, it's useful to add Labels such as `inputs.plugin_name` to your Telegraf configurations to help users filter in the UI. 
```

Comment:
Sometimes, it's handy to group Telegraf ~connfigurations~ **configuations** by specific plugins. For example, if I want to make a change to an input plugin, being able to quickly find all ~the~ Telegraf configurations using that plugin is helpful. ~To this end, it's useful to add Labels~ **Use labels** such as `inputs.plugin_name` to your Telegraf configurations to help users filter in the UI.